From 46342d6ed00efce80d4c17898d30e55b9ed39439 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20S=C5=82awi=C5=84ski?=
 <amadeuszx.slawinski@intel.com>
Date: Mon, 19 Nov 2018 16:25:44 +0100
Subject: [PATCH 236/292] ALSA: hdac: Allow HDMI Codec to have long name
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

After running:
while true; do rmmod snd_soc_skl; modprobe snd_soc_skl; done
for some time, following appears in logs

[  352.708075] HDMI HDA Codec ehdaudio10000D: hdac link not found
[  352.745778] HDMI HDA Codec ehdaudio10001D: hdac link not found
[  352.779898] HDMI HDA Codec ehdaudio10002D: hdac link not found
[  352.813913] HDMI HDA Codec ehdaudio10003D: hdac link not found
[  352.846177] HDMI HDA Codec ehdaudio10004D: hdac link not found
[  352.876984] HDMI HDA Codec ehdaudio10005D: hdac link not found
(...)

While it's probably safe to assume that no one will do this in everyday
use, fix this.

It's caused by assumption that codec name can only have 15 characters,
raise the number to 31 characters, like in snd_hda_codec_device_init.

Change-Id: Ia65c291eca42d5a14cf4698f9bdfe24656229721
Signed-off-by: Amadeusz Sławiński <amadeuszx.slawinski@intel.com>
Reviewed-on:
Reviewed-by: Rojewski, Cezary <cezary.rojewski@intel.com>
---
 sound/hda/ext/hdac_ext_bus.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/hda/ext/hdac_ext_bus.c b/sound/hda/ext/hdac_ext_bus.c
index 08cc0ce3b924..3f4f823cb0e0 100644
--- a/sound/hda/ext/hdac_ext_bus.c
+++ b/sound/hda/ext/hdac_ext_bus.c
@@ -140,7 +140,7 @@ static void default_release(struct device *dev)
 int snd_hdac_ext_bus_device_init(struct hdac_bus *bus, int addr,
 					struct hdac_device *hdev)
 {
-	char name[15];
+	char name[31];
 	int ret;
 
 	hdev->bus = bus;
-- 
2.21.0

